mito-ds/monorepo
# Sharing analyses

# Description

This PR implements this specification: https://www.notion.so/trymito/Sharing-Analyses-Specification-5acb830b2345454da6de8e71e15d25fd

major differences:
1. We store the author hash rather than the static_user id
2. Write the cell metadata rather than notebook metadata
3. Allow users to expand the steps to see what they look like before trusting them

# Testing

There are a few things to test:
1. Create an analysis, delete it off your computer in ~/.mito/saved_analyses, it should still replay
2. Create an analysis, delete the cell below, it should still replay
3. Create an analysis, delete from both locations, it should still display an error
4. Create an analysis, change your static_user_id in user.json, and it should ask you to trust it before replaying (make sure to refresh kernel)
5. Create multiple analyses in notebook
6. Create analyses in Lab, move to Notebook, and vice versa
7. Create analyses, change user.json and move one of the imported files. Should first ask you to trust, then open the PreReplayUpdateImports taskpane (and this should work). 

Any more tests you can think of as well!

# Documentation
